packages/components/src/assay/AssayGridPanel.tsx

```diff
@@ -0,0 +1,463 @@
+import React, { FC, memo, useCallback, useMemo, useState } from 'react';
+import { Filter, PermissionTypes, Utils } from '@labkey/api';
+
+import { MenuItem } from 'react-bootstrap';
+
+import { List, Map, OrderedMap } from 'immutable';
+
+import { ManageDropdownButton } from '../internal/components/buttons/ManageDropdownButton';
+import { RequiresPermission } from '../internal/components/base/Permissions';
+import { SelectionMenuItem } from '../internal/components/menus/SelectionMenuItem';
+import { MAX_EDITABLE_GRID_ROWS, NO_UPDATES_MESSAGE } from '../internal/constants';
+import { SampleActionsButton } from '../entities';
```

Comment:
hmmm, I don't think we want the subpackages importing from each other if we can avoid it. From my search, I found this usage and then the one in AsayRunDetailsPage.tsx that uses the `useSampleTypeAppContext`. Thoughts on how we can avoid this?

Reply:
Per discussion, this will get moved to the assay package.